File: mobile/src/main/java/net/mm2d/dmsexplorer/settings/theme/ThemeColorGenerator.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.settings.theme;

import android.support.annotation.ColorInt;
import android.support.annotation.NonNull;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public interface ThemeColorGenerator {
    @ColorInt
    int getIconColor(@NonNull final String title);

    @ColorInt
    int getCollapsedToolbarColor(@NonNull final String title);

    @ColorInt
    int getExpandedToolbarColor(@NonNull final String title);

    @ColorInt
    int getSubToolbarColor(@NonNull final String title);

    @ColorInt
    int getControlColor(@NonNull final String title);
}

File: mobile/src/main/kotlin/net/mm2d/dmsexplorer/settings/theme/ThemeColorGenerator.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.dmsexplorer.settings.theme

import androidx.annotation.ColorInt

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
interface ThemeColorGenerator {
    @ColorInt
    fun getIconColor(
        title: String,
    ): Int

    @ColorInt
    fun getCollapsedToolbarColor(
        title: String,
    ): Int

    @ColorInt
    fun getExpandedToolbarColor(
        title: String,
    ): Int

    @ColorInt
    fun getSubToolbarColor(
        title: String,
    ): Int

    @ColorInt
    fun getControlColor(
        title: String,
    ): Int
}
